src/cpp/statistics/fastdds/domain/DomainParticipantImpl.hpp

```diff
@@ -0,0 +1,194 @@
+     */
+    void enable_statistics_builtin_datawriters(
+            const std::string& topic_list);
+
+    /**
+     * Auxiliary function to delete the statistics builtin entities.
+     */
+    void delete_statistics_builtin_entities();
+
+    /**
+     * Auxiliary function that transforms the topic alias to the topic name.
+     * @param [in]   topic_name_or_alias string with the statistic topic name or alias.
+     * @param [out]  topic_name          string with the corresponding topic name to used.
+     * @param [out]  event_kind          statistics event kind corresponding to the topic name.
+     * @return whether the input string corresponds to a valid topic name or alias.
+     */
+    bool transform_and_check_topic_name(
+            const std::string& topic_name_or_alias,
+            std::string& topic_name,
+            EventKind& event_kind) noexcept;
+
+    /**
+     * Auxiliary function to register the statistics type depending on the statistics topic name.
+     * It also creates the topic (or finds it if already created) and returns the pointer.
+     * @param[out] topic pointer to the created topic pointer.
+     * If the method returns false the parameter is not modified.
+     * @param[in] topic_name string with the statistics topic name.
+     * @return true if successful.
+     * false in case there is incompatibility between the type associated to the Topic and the expected type.
+     */
+    bool register_statistics_type_and_topic(
+            efd::Topic** topic,
+            const std::string& topic_name) noexcept;
+
+    /**
+     * Auxiliary function that checks if the topic is already created within the domain participant.
+     * If it is not, it creates the topic and registers the type. It also checks if the type can be registered.
+     * If succesfull, it returns the pointer to the found or created topic.
+     * @param[out] topic pointer to the found or created topic pointer.
+     * @param[in] topic_name string with the topic name to find or create.
+     * @param[in] type TypeSupport to register.
+     * @return false if the topic is found but uses another type different from the expected one or if register_type
+     * fails because there is another TypeSupport using the same name. true otherwise.
+     */
+    bool find_or_create_topic_and_type(
+            efd::Topic** topic,
+            const std::string& topic_name,
+            const efd::TypeSupport& type) noexcept;
+
+    /**
+     * Auxiliary method to delete a topic and deregister a type after checking that they are consistent.
+     * @param topic_name string with the topic name.
+     * @param type_name string with the type name.
+     * @return false if the type and topic do not match or if delete_topic fails. Otherwise true.
+     */
+    bool delete_topic_and_type(
+            const std::string& topic_name) noexcept;
+
+    efd::Publisher* builtin_publisher_;
+    std::shared_ptr<DomainParticipantStatisticsListener> statistics_listener_;
```

Comment:
When building with `-DFASTDDS_STATISTICS=OFF` this line generates an error:

```
error: ‘DomainParticipantStatisticsListener’ was not declared in this scope
  176 |     std::shared_ptr<DomainParticipantStatisticsListener> statistics_listener_;
```

`DomainParticipantStatisticsListener` is only defined if `FASTDDS_STATISTICS` is set.

Reply:
That means that almost the whole file contents could be wrapped on a `#ifdef FASTDDS_STATISTICS`